src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/SharedQueueProcessingService.java

```diff
@@ -235,12 +244,12 @@ private void processBuild(LocalCIBuildJobQueueItem buildJob) {
             processingJobs.remove(buildJob.id());
             localProcessingJobs.decrementAndGet();
             addToRecentBuildJobs(finishedJob);
-            updateLocalBuildAgentInformation();
```

Comment:
Why did you remove this line? To me this seems like a change in behaviour instead of just fixing the concurrency.

Reply:
this method is already called in `addToRecentBuildJobs`, so this is just a duplicated method call